File: cobalt/frontend/src/types/entities/attributes.ts
/*
 * Copyright (C) 2026 Artorias
 * Author: Artorias
 * Repository: https://github.com/artorias-developer/cobalt
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

/**
 * Represents a server attribute.
 */
export interface AttributeEntity {
  id: number
  server_id: number
  key: string
  value: string
  created_at: string
  updated_at: string
}

File: cobalt/backend/domain/enums/attributes.py
#  Copyright (C) 2026 Artorias
#  Author: Artorias
#  Repository: https://github.com/artorias-developer/cobalt
#  SPDX-License-Identifier: AGPL-3.0-or-later

from enum import StrEnum


class AttributeSortFieldEnum(StrEnum):
    ID = "id"
    KEY = "key"
    VALUE = "value"
    CREATED_AT = "created_at"
    UPDATED_AT = "updated_at"